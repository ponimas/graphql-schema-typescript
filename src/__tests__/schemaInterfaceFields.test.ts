import { executeApiTest } from './testUtils';

describe('interface fields declaration', () => {
  it('should generate interface field resolvers', async () => {
    const schemaStr = `schema { query: Query } type Query implements IQuery { foo: String } interface IQuery { foo: String }`;
    const generated = await executeApiTest('schemaString.ts', {}, schemaStr);
    expect(generated).toEqual(expect.stringContaining(`
export interface GQLIQueryTypeResolver<TParent = any> {
  __resolveType: GQLIQueryResolveType
  foo?: IQueryToFooResolver<TParent>;
}`));
  });
});
