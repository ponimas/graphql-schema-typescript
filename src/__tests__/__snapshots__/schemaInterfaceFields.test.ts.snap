// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`interface fields declaration should generate interface field resolvers 1`] = `
"/* tslint:disable */
/* eslint-disable */
import { GraphQLResolveInfo } from 'graphql';
/**
 * This file is auto-generated by @ruiaraujo/graphql-schema-typescript
 * Please note that any changes in this file may be overwritten
 */
 

/*******************************
 *                             *
 *          TYPE DEFS          *
 *                             *
 *******************************/
export interface GQLQuery extends GQLIQuery {
  foo?: string;
}

export interface GQLIQuery {
  foo?: string;
}

/** Use this to resolve interface type IQuery */
export type GQLPossibleIQueryTypeNames = 'Query';

export interface GQLIQueryNameMap {
  IQuery: GQLIQuery;
  Query: GQLQuery;
}

/*********************************
 *                               *
 *         TYPE RESOLVERS        *
 *                               *
 *********************************/
/**
 * This interface define the shape of your resolver
 * Note that this type is designed to be compatible with graphql-tools resolvers
 * However, you can still use other generated interfaces to make your resolver type-safed
 */
export interface GQLResolver {
  Query?: GQLQueryTypeResolver;
  IQuery?: GQLIQueryTypeResolver;
}
export interface GQLQueryTypeResolver<TParent = any> {
  foo?: QueryToFooResolver<TParent>;
}

export interface QueryToFooResolver<TParent = any, TResult = any> {
  (parent: TParent, args: {}, context: any, info: GraphQLResolveInfo): TResult;
}

export interface GQLIQueryResolveType<TParent = any> {
  (parent: TParent, context: any, info: GraphQLResolveInfo): 'Query';
}
export interface GQLIQueryTypeResolver<TParent = any> {
  __resolveType: GQLIQueryResolveType
  foo?: IQueryToFooResolver<TParent>;
}

export interface IQueryToFooResolver<TParent = any, TResult = any> {
  (parent: TParent, args: {}, context: any, info: GraphQLResolveInfo): TResult;
}
"
`;
